backends/concrete-cpu/implementation/src/c_api/fheint.rs

```diff
@@ -0,0 +1,180 @@
+use tfhe::integer::ciphertext::Expandable;
+use tfhe::integer::IntegerCiphertext;
+use tfhe::shortint::parameters::{Degree, NoiseLevel};
+use tfhe::shortint::{CarryModulus, Ciphertext, MessageModulus};
+use tfhe::FheUint8;
+
+#[repr(C)]
+pub struct TfhersFheIntDescription {
+    width: usize,
+    is_signed: bool,
+    lwe_size: usize,
+    n_cts: usize,
+    degree: usize,
+    noise_level: usize,
+    message_modulus: usize,
+    carry_modulus: usize,
+    ks_first: bool,
+}
+
+impl TfhersFheIntDescription {
+    fn zero() -> TfhersFheIntDescription {
+        TfhersFheIntDescription {
+            width: 0,
+            is_signed: false,
+            lwe_size: 0,
+            n_cts: 0,
+            degree: 0,
+            noise_level: 0,
+            message_modulus: 0,
+            carry_modulus: 0,
+            ks_first: false,
+        }
+    }
+
+    /// Create a `Ciphertext` from the lwe and `self` metadata
+    fn ct_from_lwe(&self, lwe: LweCiphertext<Vec<u64>>) -> Ciphertext {
+        Ciphertext::new(
+            lwe,
+            // TODO: how to set degree here?
+            Degree::new(self.degree),
+            // TODO: how to set noise here?
+            NoiseLevel::UNKNOWN,
+            MessageModulus(self.message_modulus),
+            CarryModulus(self.carry_modulus),
+            if self.ks_first {
+                PBSOrder::KeyswitchBootstrap
+            } else {
+                PBSOrder::BootstrapKeyswitch
+            },
+        )
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn concrete_cpu_tfhers_uint8_description(
+    serialized_data_ptr: *const u8,
+    serialized_data_len: usize,
+) -> TfhersFheIntDescription {
+    nounwind(|| {
+        // deserialize fheuint8
+        let mut serialized_data = Cursor::new(slice::from_raw_parts(
+            serialized_data_ptr,
+            serialized_data_len,
+        ));
+        let fheuint: FheUint8 = match bincode::deserialize_from(&mut serialized_data) {
+            Ok(value) => value,
+            Err(_) => {
+                return TfhersFheIntDescription::zero();
+            }
+        };
+        // get metadata from fheuint's ciphertext
+        let (radix, _) = fheuint.into_raw_parts();
+        let blocks = radix.blocks();
+        let ct = match blocks.first() {
+            Some(value) => &value.ct,
+            None => {
+                return TfhersFheIntDescription::zero();
+            }
+        };
+        TfhersFheIntDescription {
+            width: 8,
+            is_signed: false,
+            lwe_size: ct.lwe_size().0,
+            n_cts: blocks.len(),
+            degree: blocks[0].degree.get(),
+            noise_level: blocks[0].noise_level().get(),
+            message_modulus: blocks[0].message_modulus.0,
+            carry_modulus: blocks[0].carry_modulus.0,
+            ks_first: blocks[0].pbs_order == PBSOrder::KeyswitchBootstrap,
+        }
+    })
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn concrete_cpu_tfhers_uint8_to_lwe_array(
```

Comment:
I understand you mean with the get description function. This could be done, but the two functions are actually unrelated. With recent changes, the get description function is no longer necessary, but it could be used to get some information for debugging for example. So I don't think we should merge them.

Reply:
You can see how it's used in my following push

Reply:
Ok but should be used at least to verify that the imported ciphertext parameters are equals to ones given by the developer, wdyt?